subgraphs/notional-finance/src/setters/financialMetrics.ts

```diff
@@ -26,55 +25,64 @@ import {
 } from "../getters/market";
 import { getOrCreateLendingProtocol } from "../getters/protocol";
 import { getOrCreateToken } from "../getters/token";
+import { getOrCreateInterestRate } from "../getters/interestRate";
+import { MarketDailySnapshot } from "../../generated/schema";
 
 export function updateFinancials(
   event: ethereum.Event,
-  amountUSD: BigDecimal,
-  marketId: string
+  marketId: string,
+  prevMarketDailySnapshot: MarketDailySnapshot
 ): void {
-  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
-  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
-  const marketDailySnapshot = getOrCreateMarketDailySnapshot(event, marketId);
-  const protocol = getOrCreateLendingProtocol();
   const market = getOrCreateMarket(event, marketId);
+  const protocol = getOrCreateLendingProtocol();
+  const marketHourlySnapshot = getOrCreateMarketHourlySnapshot(event, marketId);
+  const financialsDailySnapshots = getOrCreateFinancialsDailySnapshot(event);
 
-  // fees and revenue amounts
-  const feesUSD = amountUSD.times(NOTIONAL_TRADE_FEES);
-  const totalRevenueUSD = feesUSD;
-  const supplySideRevenueUSD = feesUSD.times(
-    NOTIONAL_SUPPLY_SIDE_REVENUE_SHARE
+  const dailyId = event.block.timestamp.toI64() / SECONDS_PER_DAY;
+  const marketDailySnapshot = getOrCreateMarketDailySnapshot(
+    event,
+    dailyId,
+    marketId
   );
-  const protocolSideRevenueUSD = feesUSD.times(
-    NOTIONAL_PROTOCOL_SIDE_REVENUE_SHARE
+
+  // total/cumulative revenue calculation (at event time)
+  const dailyLendingRate = getOrCreateInterestRate(market.id).rate.div(
```

Comment:
Going off my last comment, we should not assume the last snapshot was 1 day ago. If it is longer, we should normalize to that.

Reply:
Agreed, addressing this in my recent changes.

Reply:
Agreed, addressing this in my recent changes.